admin-dev/themes/new-theme/js/pages/product/components/combination-list/bulk-delete-handler.ts

```diff
@@ -69,43 +69,80 @@ export default class BulkDeleteHandler {
     }
 
     bulkDeleteBtn.addEventListener('click', () => {
-      try {
-        const selectedCombinationsCount = this.bulkChoicesSelector.getSelectedCheckboxes().length;
-        const confirmLabel = bulkDeleteBtn.dataset.modalConfirmLabel
-          ?.replace(/%combinations_number%/, String(selectedCombinationsCount));
-
-        const modal = new ConfirmModal(
-          {
-            id: 'modal-confirm-delete-combinations',
-            confirmTitle: bulkDeleteBtn.innerHTML,
-            confirmMessage: bulkDeleteBtn.dataset.modalMessage,
-            confirmButtonLabel: confirmLabel,
-            closeButtonLabel: bulkDeleteBtn.dataset.modalCancelLabel,
-            closable: true,
-          },
-          async () => {
-            const response = await this.bulkDelete();
-            $.growl({message: response.message});
-            this.eventEmitter.emit(CombinationEvents.refreshCombinationList);
-          },
-        );
-        modal.show();
-      } catch (error) {
-        const errorMessage = error.response?.JSON ?? error;
-        $.growl.error({message: errorMessage});
-      }
+      this.bulkChoicesSelector.getSelectedIds().then((selectedCombinationIds) => {
```

Comment:
```suggestion
      const selectedCombinationIds = await this.bulkChoicesSelector.getSelectedIds();
```

Reply:
I prefer this notation here, it emphasizes the fact that the method process wait for the resolution of `getSelectedIds` It's not an asynchronous call

Reply:
Did you try with an `await` ? Maybe it doesn't work?

Reply:
yes i tried, but "'await' expressions are only allowed within async functions and at the top levels of modules.", so it seems i cannot use it inside event listener callback (idk if there is other way than using `.then()`